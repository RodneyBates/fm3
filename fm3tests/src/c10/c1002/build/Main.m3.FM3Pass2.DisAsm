 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L1(16_1),[1,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenScopeLt(L1(16_1))
 ItkOpenDeclListLt(L1(16_1),[3,2])
 ItkProcDefLt(Id5"P",[3,2])
 ItkDeclNo(D12,[3,12])
 ItkDeclScopeLt(L2(16_2))
 ItkSignatureProperLt([4,4])
 ItkFormalsListLt(L4(16_4),[4,6])
 ItkVALUEFormalLt([4,6])
 ItkVALUEFormalIdListLt(L1(16_1),[4,6])
 ItkDeclNo(D4,[4,6])
 ItkVALUEFormalIdListRt(L1(16_1),[4,6])
 ItkVALUEFormalType([4,8])
 ItkDefTopExprNo(L8(16_8),[4,10])
 ItkVALUEFormalValue([5,4])
 ItkDeclValAbsent([5,4])
 ItkVALUEFormalRt([4,6])
 ItkFormalsListSep(L1(16_1),[5,4])
 ItkVALUEFormalLt([5,6])
 ItkVALUEFormalIdListLt(L1(16_1),[5,12])
 ItkDeclNo(D5,[5,12])
 ItkVALUEFormalIdListRt(L1(16_1),[5,12])
 ItkVALUEFormalType([5,17])
 ItkDefTopExprNo(L7(16_7),[5,19])
 ItkVALUEFormalValue([6,4])
 ItkDeclValAbsent([6,4])
 ItkVALUEFormalRt([5,6])
 ItkFormalsListSep(L2(16_2),[6,4])
 ItkROFormalLt([6,6])
 ItkROFormalIdListLt(L1(16_1),[6,15])
 ItkDeclNo(D6,[6,15])
 ItkROFormalIdListRt(L1(16_1),[6,15])
 ItkROFormalType([6,19])
 ItkDefTopExprNo(L6(16_6),[6,21])
 ItkROFormalValue([7,4])
 ItkDeclValAbsent([7,4])
 ItkROFormalRt([6,6])
 ItkFormalsListSep(L3(16_3),[7,4])
 ItkVARFormalLt([7,6])
 ItkVARFormalIdListLt(L1(16_1),[7,10])
 ItkDeclNo(D7,[7,10])
 ItkVARFormalIdListRt(L1(16_1),[7,10])
 ItkVARFormalType([7,15])
 ItkDefTopExprNo(L5(16_5),[7,17])
 ItkVARFormalValue([7,25])
 ItkDeclValAbsent([7,25])
 ItkVARFormalRt([7,6])
 ItkFormalsListRt(L4(16_4),[4,6])
 ItkDeclScopeRt(L2(16_2))
 ItkResultTypeAbsent([9,2])
 ItkRaisesSetLt(L0(16_0),[9,2])
 ItkRaisesSetRt(L0(16_0),[9,2])
 ItkSignatureProperRt([4,4])
 ItkProcDefBody([9,2])
 ItkDeclScopeLt(L3(16_3))
 ItkOpenScopeLt(L3(16_3))
 ItkOpenDeclListLt(L2(16_2),[9,4])
 ItkConstDeclLt([9,10])
 ItkDeclNo(D10,[9,10])
 ItkConstDeclType([9,13])
 ItkDeclTypeAbsent([9,13])
 ItkConstDeclValue([9,13])
 ItkConstDeclRt([9,10])
 ItkOpenDeclListSep(L1(16_1),[9,4])
 ItkTypeDeclLt([11,9])
 ItkDeclNo(D11,[11,9])
 ItkTypeDeclType([12,4])
 ItkDeclScopeLt(L4(16_4))
 ItkFieldDeclListLt(L2(16_2),[13,8])
 ItkFieldDeclLt([13,8])
 ItkFieldDeclIdListLt(L1(16_1),[13,8])
 ItkDeclNo(D8,[13,8])
 ItkFieldDeclIdListRt(L1(16_1),[13,8])
 ItkFieldDeclType([13,8])
 ItkDefTopExprNo(L4(16_4),[13,14])
 ItkFieldDeclValue([13,12])
 ItkDeclValAbsent([14,6])
 ItkFieldDeclRt([13,8])
 ItkFieldDeclListSep(L1(16_1),[14,6])
 ItkFieldDeclLt([14,8])
 ItkFieldDeclIdListLt(L1(16_1),[14,8])
 ItkDeclNo(D9,[14,8])
 ItkFieldDeclIdListRt(L1(16_1),[14,8])
 ItkFieldDeclType([14,8])
 ItkDefTopExprNo(L3(16_3),[14,14])
 ItkFieldDeclValue([14,12])
 ItkDeclValAbsent([15,6])
 ItkFieldDeclRt([14,8])
 ItkFieldDeclListRt(L2(16_2),[13,8])
 ItkDeclScopeRt(L4(16_4))
 ItkDefTopExprNo(L2(16_2),[12,6])
 ItkTypeDeclRt([11,9])
 ItkOpenDeclListRt(L2(16_2),[9,4])
 ItkStmtListLt(L0(16_0),[18,6])
 ItkBecomesLt([18,8])
 ItkBecomesInfix([18,8])
 ItkDefTopExprNo(L1(16_1),[18,11])
 ItkBecomesRt([18,8])
 ItkStmtListRt(L0(16_0),[18,6])
 ItkOpenScopeRt(L3(16_3))
 ItkDeclScopeRt(L3(16_3))
 ItkProcDefRt(Id5"P",[19,8])
 ItkOpenDeclListRt(L1(16_1),[3,2])
 ItkModuleBegin(L1(16_1),[21,2])
 ItkStmtListLt(L0(16_0),[22,2])
 ItkStmtListRt(L0(16_0),[22,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L1(16_1),[1,0])
 ItkRightEnd()
 ItkBOF()
     0
