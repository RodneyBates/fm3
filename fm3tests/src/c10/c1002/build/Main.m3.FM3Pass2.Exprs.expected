From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [12,6]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = type
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype Expr1ScopeTyp: *)
 ExpScopeRef1 = NIL
 (* From subtype ExprRecTypeTyp: *)


From unit expression map, Expr No 2 16_****************
 ExpStackLink = ExprNo 1 at 16_**************** [12,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [14,14]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 3 16_****************
 ExpStackLink = ExprNo 1 at 16_**************** [12,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 3
 ExpPosition = [13,14]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 4 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 4
 ExpPosition = [7,17]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 5 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 5
 ExpPosition = [6,21]
 ExpOpcode = LONGINT
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 6 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 6
 ExpPosition = [5,19]
 ExpOpcode = CHAR
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 7 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 7
 ExpPosition = [4,10]
 ExpOpcode = BOOLEAN
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


